File: BurningKnight/Assets/Assets.cs
using System.Collections.Generic;
using Microsoft.Xna.Framework.Content;

namespace BurningKnight.assets
{
  public static class Assets
  {
    public static ContentManager content;
    public static Mods Mods { get; private set; }

    private static readonly List<AssetManager> managers = new List<AssetManager>();
    
    public static void Load()
    {
      managers.Add(new Graphics());
      managers.Add(new Audio());
      managers.Add(Mods = new Mods());

      foreach (AssetManager manager in managers)
      {
        manager.LoadAssets();
      }
    }

    public static void Destroy()
    {
      foreach (AssetManager manager in managers)
      {
        manager.Destroy();
      }
    }
  }
}

File: core/src/org/rexcellentgames/burningknight/assets/Assets.java
package org.rexcellentgames.burningknight.assets;

import com.badlogic.gdx.assets.AssetManager;

public class Assets {
	public static AssetManager manager;
	private static boolean done;
	public static boolean finishedLoading;

	public static void init() {
		manager = new AssetManager();

		Audio.targetAssets();
		Graphics.targetAssets();
	}

	public static boolean updateLoading() {
		boolean val = manager.update();

		if (val && !done) {
			done = true;
			finishedLoading = true;
			loadAssets();
		}

		return val;
	}

	public static void finishLoading() {
		finishedLoading = true;
		manager.finishLoading();
		loadAssets();
	}

	public static void loadAssets() {
		Audio.loadAssets();
		Graphics.loadAssets();
	}

	public static void destroy() {
		Graphics.destroy();
		Audio.destroy();

		if (manager != null) {
			manager.dispose();
		}
	}
}